mapbox/services/analytics.py

```diff
@@ -0,0 +1,54 @@
+from mapbox.services.base import Service
+import dateutil.parser
+from dateutil.relativedelta import *
+from uritemplate import URITemplate
+
+class Analytics(Service):
+    """Access to Analytics API"""
+
+    valid_resource_types = ['tokens', 'styles', 'accounts', 'tilesets']
+
+    @property
+    def baseuri(self):
+        return 'https://{0}/analytics/v1'.format(self.host)
+
+    def _validate_resource_type(self, resource_type):
+        if resource_type not in self.valid_resource_types:
+            raise ValueError("{0} is not a valid profile".format(resource_type))
+        return resource_type
+
+    def _validate_period(self, period):
```

Comment:
Better would be `def _validate_period(self, start, end):` because of clearer semantics: we need exactly one start value and one end value, a 1-length list or 3+ length list is no good.